code/datums/flock/flock.dm

```diff
@@ -275,24 +275,29 @@
 		dummy.add_filter("outline", 1, outline_filter(size=1,color="#00ff9d"))
 		target.vis_contents += dummy
 
+		play_animation()
+
 		SPAWN(0)
 			while(TIME < src.end_time)
-				animate(dummy, time = duration/9, alpha = 100)
-				for (var/i in 1 to 4)
-					animate(time = duration/9, alpha = 255)
-					animate(time = duration/9, alpha = 100)
-				sleep(duration)
+				var/delta = src.end_time - TIME
+				sleep(min(src.duration, delta))
```

Comment:
why not just spawn for this in the first place lmao - you changed the whole semantics of this, was that intentional?

Reply:
Because the end_time gets modified when it's re-pinged, so it has to sleep for the initial duration and then check if it should sleep for longer. I'm sorry, I don't see how you could do this with just a spawn?
![image](https://user-images.githubusercontent.com/20713227/163360402-a4267c4a-537f-4135-95a4-88ac779f809f.png)